functions/src/aqi-calculation/purple-air-response.ts

```diff
@@ -139,69 +139,45 @@ async function purpleAirToFirestore(): Promise<void> {
     }
 
     // Add readings to the PM 2.5 buffer
+    const sensorDocUpdate = Object.create(null);
+
+    // Only add the most recent reading data if a new reading exists
+    if (reading) {
```

Comment:
If a new reading does not exist, we still want to add a default buffer element (NaNs/nones). I think we need to add an else where we don't update the last reading time and stuff but still add an element to the buffer and increment the index.

Reply:
This still happens--the `sensorDocUpdate` variable only controls what is in the sensor doc, not the buffer, which is updated in later lines. I updated this code to make this clearer though

Reply:
That was definitely my bad, I thought it was all inside the if statement